File: customer-module/backend/src/main/java/com/pos/customer/service/ProductService.java
package com.pos.customer.service;

import org.springframework.http.ResponseEntity;

import com.pos.customer.dto.ProductListDto;
import com.pos.customer.entity.Product;

public interface ProductService {
	String baseUrl="https://admin-pos.herokuapp.com/api/";
	
	ResponseEntity<ProductListDto> getProducts() throws Exception;
	ResponseEntity<ProductListDto> getCategoryProducts(Integer categoryId) throws Exception;
	ResponseEntity<Product> getProduct(Integer productId) throws Exception;
}

File: customer-module/frontend/src/app/product.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root',
})
export class ProductService {
  private baseUrl = 'https://pos-customer.herokuapp.com/customer/';

  constructor(private httpClient: HttpClient) {}

  public getProducts(): Observable<any> {
    return this.httpClient.get(this.baseUrl + 'products');
  }
  public getCategoryProducts(categoryId): Observable<any> {
    return this.httpClient.get(this.baseUrl + '/proudcts/' + categoryId);
  }
  public getProduct(productId): Observable<any> {
    return this.httpClient.get(this.baseUrl + 'product/' + productId);
  }
}
